python/cudf/cudf/core/index.py

```diff
@@ -1560,10 +1563,7 @@ def __contains__(self, item):
             return False
         if not item % 1 == 0:
             return False
-        if self._start <= item < self._stop:
-            return True
-        else:
-            return False
+        return item in range(self._start, self._stop, self._step)
```

Comment:
This could get very expensive no? I think we should be able to calculate this by plain math no?

Reply:
I think this should be efficient since it's not really looping, but using the `__contains__` method of `range`. Should be efficient (and possibly more so than implementing the math ourselves in Python).

Reply:
I see, was under an assumption it probably is looping. You are right, sorry for the confusion here. Range `__contains__` appears to be already doing that:

```python
    def __contains__(self, num):
        if self.step < 0:
            if not (self.stop < num <= self.start):
                return False
        else:
            if not (self.start <= num < self.stop):
                return False
        return (num - self.start) % self.step == 0
```